apps/web-client/src/lib/utils/misc.ts

```diff
@@ -40,3 +40,34 @@ export function normalizeName(name: string) {
 		.join(" ") // Rejoin with single spaces
 		.trim();
 }
+
+export function matchesName(needle: string, haystack: string) {
+	// Helper to normalize and tokenize the needle string into conditions
+	const getConditions = (str: string) => {
+		return str
+			.toLowerCase()
+			.replace(/[^\w\s]/g, " ") // Replace punctuation with a space
+			.replace(/\s+/g, " ") // Collapse multiple spaces into one
+			.trim()
+			.split(" ")
+			.filter((part) => part.length > 0); // Remove any empty parts resulting from normalization
+	};
+
+	const needleConditions = getConditions(needle);
+
+	// If after normalization the needle has no conditions (e.g., needle was empty or just punctuation)
+	if (needleConditions.length === 0) {
+		return needle.trim().length === 0;
```

Comment:
What is the reasoning behind this? Wy not returning `true` when `needleConditions` has length 0?

As far as I can tell, it makes these two queries yield different results: `   ` (only whitespace) and `  . ` (possibly whitespace and punctuation).